File: LOO 2°Parcial/Tarea 3/Jefe.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EmpleadosCSharp
{
    class Jefe : Empleado
    {
        protected double salarioSemanal;
        public Jefe() : base()
        {
            this.salarioSemanal = 0;
        }
        public Jefe(String nombre, double salario) : base(nombre)
        {
            this.salarioSemanal = salario > 0 ? salario : 0;
        }
        public double SalarioSemanal
        {
            set { this.salarioSemanal = value > 0 ? value : 0; }
            get { return this.salarioSemanal; }
        }
        public override double pagoDiario()
        {
            return this.salarioSemanal / 5;
        }
        public override void imprimir()
        {
            Console.WriteLine("Empleado tipo JEFE");
            Console.WriteLine("NOMBRE: " + this.nombre + "\nSALARIO SEMANAL: " + this.salarioSemanal + "\nPAGO DIARIO: " + this.pagoDiario());
        }
    }
}

File: LOO 2°Parcial/PY's/Jefe.py
# -*- coding: utf-8 -*-
"""
Created on Fri Mar 19 12:23:32 2021

@author: MedNo
"""

import Empleado as Emp

class Jefe(Emp.Empleado):
    def __init__(self, nombre="", salario=0):
        Emp.Empleado.__init__(self,nombre)
        if(salario>0):
            self.__salarioSemanal = salario
        else:
            self.__salarioSemanal = 0
            
    @property
    def salarioSemanal(self):
        return self.__salarioSemanal
    
    @salarioSemanal.setter
    def salarioSemanal(self, value):
        if(value>0):
            self.__salarioSemanal = value
        else:
            self.__salarioSemanal = 0
            
    def pagoDiario(self):
        return self.__salarioSemanal / 5
    
    def imprimir(self):
        print("Empleado tipo JEFE");
        print("NOMBRE: " + self.nombre + "\nSALARIO SEMANAL: " + str(self.__salarioSemanal) + "\nPAGO DIARIO: " + str(self.pagoDiario()))
        
      